src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -241,12 +248,17 @@ public void Close()
     /// <inheritdoc/>
     public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk)
     {
+        // Currently, TestRequestSender always passes null for lastChunk in case of an aborted
+        // discovery but we are not making this assumption here to ease potential future
+        // evolution.
+        _discoveryDataAggregator.MarkSourcesBasedOnDiscoveredTestCases(lastChunk, isComplete: !discoveryCompleteEventArgs.IsAborted, ref _previousSource);
```

Comment:
this is per proxy, and marks it as complete, which I think we make all sources complete, even the ones we don't own.